### DISCUSSION TITLE
Use escape key to do another function

### DISCUSSION THREAD

**Discussion by gamellis** [faceless-ui/modal#44] [Q&A]
Hey @jacobsfletch,

Thanks for fixing that escape key bug last year. 😄 Here's my setup for controlling the modal.

```
const [renderModal, setRenderModal] = useState(false)
const modalSlug = '...'

useEffect(() => {
    if (renderModal) {
        toggleModal(modalSlug)
    }
}, [renderModal, modalSlug, toggleModal])
```

I've got a `handleDismissModal` to close the modal with an extra `myFunction()`

```
const handleDismissModal = () => {
    setRenderModal(false)
    closeModal(modalSlug)
    myFunction()
}
```

When the escape key is pressed, my `handleDismissModal` function isn't used (of course). I read through the modal source code and couldn't figure out how to access the key press event to pass in a function.

I guess I am missing something about how the modal component "should be" used.

Thanks for all your help!

**Comment by jacobsfletch** [comment]
@gamellis thank you for the support 🙌 ! Is your `handleDismissModal` callback defined _within_ your modal? Because in that case, the modal is probably unmounted and never has a chance to execute anything after `closeModal()`. Can you run your custom function _before_ closing the modal? Otherwise, lift you callback into a component higher in the tree that doesn't unmount when the modal is closed. Hope this helps! I'm happy to continue troubleshooting here.
